src/components/main.js

```diff
@@ -2,27 +2,52 @@ import { useState } from "react";
 
 export default function Header() {
   const [result, setResult] = useState(null);
+  const [file, setFile] = useState(null);
 
   const handleUpload = () => {
-    // TODO: pass input to the model
-    fetch("http://localhost:5000/result")
-      .then(res => res.json())
-      .then(data => {
-        setResult(data.result);
-      });
+    if (file) {
+      const data = new FormData();
+      data.append("file", file);
+      data.append("filename", file.name);
+
+      // TODO: Eventually can get actual data not just filename. Encrypt it
+      fetch("http://localhost:5000/result", {
```

Comment:
Maybe extract URL as global variable so we can change as needed. localhost:5000 is good for now!

Reply:
Good suggestion!

Reply:
Good suggestion!